### PR TITLE
Remove Jumpstart

### LINKED ISSUE #12601: Remove recommended features prompt (aka Jumpstart)
Once security and performance checklists shown during the onboarding flow are fully implemented, jumpstart becomes redundant and can be removed.

Screenshot 2019-06-07 at 11 35 01

### PR DESCRIPTION
**Pull request by simison**
Removing everything jumpstart related.

~Basically informs me that I should first PR a change which just modifies minimum to hide Jumpstart from Jetpack, and then PR separately follow-ups to clean up the code.~ Got convinced by @dereksmart that this is solid. (p1560894750228800-slack-jetpack-dna). If further testing reveals surprises that take more than a day to fix, we can reassess and do the simple hiding instead.

Jumpstart might normally get disabled once some module toggles are changed. (Will still need to confirm if this is the case). If so, should we keep Jumpstart for those who didn't activate any features in the checklist, @jeffgolenski?

See inline comments for some todos.

Resolves https://github.com/Automattic/jetpack/issues/12601

#### Changes proposed in this Pull Request:
* Remove Jumpstart feature

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Removes existing feature.

#### Testing instructions:
- Create a new Jetpack site and connect it
- From Calypso checklist, click back to wp-admin without changing settings — you shouldn't see jumpstart and no errors, observe also PHP log.
- With another site, when in Calypso, turn on some features in settings and then return to wp-admin. Still no jumpstart and no errors. 
- Try different plan tiers

#### Proposed changelog entry for your changes:
- Remove feature to activate recommended features with one click

### CODE REVIEW

File: .eslintignore
Comment (simison) [inline .eslintignore]: ES5 file that was eslint-erroring a whole lotta. :-)

File: composer.lock
Comment (simison) [inline composer.lock]: Revert these changes before merge.

File: class.jetpack.php
Comment (simison) [inline class.jetpack.php]: Need to investigate what was going on here and if `jumpstart` option is used to somehow detect the state of Jetpack, not just state of Jumpstart?

**Comment by simison** [comment]
Need to investigate what's up with this:
https://github.com/Automattic/jetpack/blob/f74b64b2cf2f7337cae0d25c5017129f52c083d6/tools/build-module-headings-translations.php#L39

File: packages/options/legacy/class.jetpack-options.php
Comment (simison) [inline packages/options/legacy/class.jetpack-options.php]: Noting that I got heaps of phpcs errors from this file and had to use `--no-verify` to skip tests and commit:

```
FILE: ...mattic/jetpack/packages/options/legacy/class.jetpack-options.php
----------------------------------------------------------------------
FOUND 39 ERRORS AND 10 WARNINGS AFFECTING 41 LINES
----------------------------------------------------------------------
....
```

**Comment by simison** [comment]
I'd like early testing/reviews on this even tho I'd still consider it also to be in progress.

**Review by sirreal — DISMISSED** [review]
This seems to work fine and some quick digging around doesn't reveal any leftover jumpstart code, although I'm sure there's some 🙂

I do think we need some careful consideration:

- What's enabled by default?
- What would have been enabled by jumpstart?
- What would be covered by the new checklist?
- How does the following the checklist differ from triggering jumpstart?

I'd like to have a clear understanding of the above written out so we can refer to this PR and have clear expectations.

**Review by jeherve — DISMISSED** [review]
image

❤️ 

This looks good to me. I'd be tempted to merge it sooner rather than later, so we can test it with all future PR reviews in the next few days.

**Comment by dereksmart** [comment]
This tested well for me too. Would be happy to merge after resolving conflicts.

File: composer.lock
Comment (simison) [reply composer.lock]: done https://github.com/Automattic/jetpack/pull/12747/commits/a12a454f3d91eae12008b43fa1751e1edefb38eb

**Comment by simison** [comment]
Rebased & removed lockfile changes

**Comment by simison** [comment]
>What's enabled by default?
>What would have been enabled by jumpstart?
>What would be covered by the new checklist?

Here's a list of tasks in the checklist — I'm adding ✅to those that were also enabled by Jumpstart:
- Downtime Monitoring ✅
- Automatic Plugin Updates
- WordPress.com sign in ✅
- Site Accelerator ✅
- Lazy Load Images ✅
- Video Hosting
- Enhanced Search

Things that were always auto-enabled regardless:
- Brute force login attack protection
- Backup and Scan
- Spam filtering

**Review by jeherve — APPROVED** [review]
This still looks good to me. Merging.

**Comment by lezama** [comment]
+16 −820  ❤️

**Pull request merged**
